gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/factories/GaaSGraphsFactory.java

```diff
@@ -75,6 +82,24 @@ private static RestApiStatus getStatus(final Gaffer gaffer) {
         return gaffer.getStatus() != null && gaffer.getStatus().getProblems() != null ? gaffer.getStatus().getProblems() : new ArrayList<String>();
     }
 
+    private static String getStoreType(final Gaffer gaffer) {
```

Comment:
This is exactly what we wanted to move away from - having all of the store types hard coded...

Reply:
Sure totally agree about the hardcoded strings - I have refactored them to use enums in another pull request #95 instead or any String or non-typed constants. 

This requirement was to add the storeType to the API response to GET Graphs response body - would you prefer to use the Store.class name as store type that comes from the StoreProperties 'store class name' instead and just have the API return whatever value was set for that property? 

If not, please could you share how you want it to be implemented?
- Where the API can get the store class used
- How it should be handled in the API (Should it just be treated as any String throughout the API and returned to the UI/Client to handle) ?

Reply:
```
@Autowired
public List<StoreType> storeTypes
...
for (final StoreType store : storeTypes) {
    if (storeClass.equalIgnoreCase(store.getName()) {
        return store.getName;
    }
}
return "n/a"
```

Reply:
And I don't need to do any looping above

Reply:
The difference is you do not need to update an Enum every time you add a new store type. There are new store types planned so I am future proofing.

Reply:
I know what you mean - but you'll still have to seperately add some additional validation in the presentation layer for seperation of concerns as well and for other parts of the code to know what has StoreType has been requested - so may as well just have it as an enum and have the added benefit of type safety for developers developing the code as it is painful using 'any Strings' everywhere when referencing a type - so that is making it more unmaintainable - I'm sure Joshua Bloch in Core Java prefers enums over constants when referencing types.

Developers will have to add a new factory for new store types anyway so the method `StoreType getType()` will let them know a new type will have to exist for the code that the factory and other areas of the code can use, so adding one additional enum is really easy to do for a developer